src/UI/Implementation/Component/Dropzone/File/Renderer.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace ILIAS\UI\Implementation\Component\Dropzone\File;
+
+use ILIAS\UI\Component\Component;
+use ILIAS\UI\Implementation\DefaultRenderer;
+use ILIAS\UI\Implementation\Render\AbstractComponentRenderer;
+use ILIAS\UI\Implementation\Render\ResourceRegistry;
+
+/**
+ * Class Renderer
+ *
+ * Renderer implementation for file dropzones.
+ *
+ * @author  nmaerchy <nm@studer-raimann.ch>
+ * @author  Stefan Wanzenried <sw@studer-raimann.ch>
+ *
+ * @package ILIAS\UI\Implementation\Component\Dropzone\File
+ */
+class Renderer extends AbstractComponentRenderer {
+
+	/**
+	 * @var $renderer DefaultRenderer
+	 */
+	private $renderer;
+
+
+	/**
+	 * @inheritdoc
+	 */
+	protected function getComponentInterfaceName() {
+		return array(
+			\ILIAS\UI\Component\Dropzone\File\Standard::class,
+			\ILIAS\UI\Component\Dropzone\File\Wrapper::class,
+		);
+	}
+
+
+	/**
+	 * @inheritdoc
+	 */
+	public function render(Component $component, \ILIAS\UI\Renderer $default_renderer) {
+		$this->checkComponent($component);
+		$this->renderer = $default_renderer;
+		if ($component instanceof \ILIAS\UI\Component\Dropzone\File\Wrapper) {
+			return $this->renderWrapper($component);
+		}
+		if ($component instanceof \ILIAS\UI\Component\Dropzone\File\Standard) {
+			return $this->renderStandard($component);
+		}
+	}
+
+
+	/**
+	 * @inheritDoc
+	 */
+	public function registerResources(ResourceRegistry $registry) {
+		parent::registerResources($registry);
+		$registry->register("./src/UI/templates/js/libs/jquery.dragster.js");
```

Comment:
This is a small helper JS file to handle the drag/drop events more easily (which are pretty messy). It is not part of fine uploader nor available over npm.